### PR TITLE
Pin django-storages extra requirements

### PR DESCRIPTION
**Pull request by gerrod3**
The latest django-storages 1.14 adds new fields to S3 config that are not accounted for in the domain serializer, so this requirement should have been pinned from the first introduction of domains. This will need to be backported.

### CODE REVIEW

**Comment by gerrod3** [comment]
Using plugin-template: https://github.com/pulp/plugin_template/pull/787

**Comment by gerrod3** [comment]
I've confirmed that the CI now respects the bounds for S3 and Azure by looking at the pip logs, but I can't figure out why the CI still installs django-storages even for the Pulp and Lowerbounds runners.

**Comment by mdellweg** [comment]
> I've confirmed that the CI now respects the bounds for S3 and Azure by looking at the pip logs, but I can't figure out why the CI still installs django-storages even for the Pulp and Lowerbounds runners.

They are installed in the containerfile which is the same for all test scenarios.

**Comment by gerrod3** [comment]
Waiting for the weekly CI update job to run before rebasing now that the plugin template PR has been merged.

**Review by lubosmj — APPROVED** [review]

File: setup.py [hunk 1]
```diff
@@ -21,9 +21,9 @@
     install_requires=requirements,
     extras_require={
         "sftp": ["django-storages[sftp]"],
-        "s3": ["django-storages[boto3]"],
+        "s3": ["django-storages[boto3]<=1.13.2"],
         "google": ["django-storages[google]>=1.13.2"],
```
Comment (mdellweg) [inline setup.py:25]: Shouldn't we add something here too?

Comment (gerrod3) [reply setup.py:25]: For google you mean? Maybe, I didn't since this change was specific for domains and domains doesn't support google backends yet.

Comment (mdellweg) [reply setup.py:25]: Right. But in general, i think these are as good dependencies as all the others. And we should add the upper bound in the same manner. (Can be a separate PR if you want.)

**Review by mdellweg — APPROVED** [review]

**Comment by mdellweg** [comment]
Do you plan to up the dependencies an include the new fields in the domain config? At least please track it with an issue.

**Comment by gerrod3** [comment]
> Do you plan to up the dependencies an include the new fields in the domain config? At least please track it with an issue.

Yes I plan to add the new fields when I will update the dependencies. https://github.com/pulp/pulpcore/issues/4592

**Pull request merged**